tempest/api/workloadmgr/restore/test_exclude_cinder_volume.py

```diff
@@ -0,0 +1,444 @@
+            self.data_ops_on_mount_point(floating_ip_1, mount_point[0], 3)
+            LOG.debug("Created disk and mounted the attached volume")
+
+            # calculate checksum
+            md5sums_before_full = {}
+            md5sums_before_full = self.calcmd5sum(
+                floating_ip_1, mount_point[0])
+            LOG.debug("MD5sums for directory on original vm : " +
+                      str(md5sums_before_full))
+
+            # update volume metadata
+            body = self.modify_volume_metadata(
+                volume_id, tvaultconf.enable_cinder_volume_exclusion)
+            LOG.debug("Response to update volume metadata req: " + str(body))
+
+            # Create workload using CLI
+            workload_create = command_argument_string.workload_create + \
+                              " --instance instance-id=" + str(vm_id)
+
+            rc = cli_parser.cli_returncode(workload_create)
+
+            if rc != 0:
+                reporting.add_test_step(
+                    "Execute workload-create command using cli",
+                    tvaultconf.FAIL)
+                raise Exception(
+                    "Command workload create did not execute correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload-create command using cli",
+                    tvaultconf.PASS)
+
+            time.sleep(10)
+
+            # check workload status
+            wid = query_data.get_workload_id_in_creation(tvaultconf.workload_name)
+            LOG.debug("Workload ID: " + str(wid))
+            if (wid is not None):
+                self.wait_for_workload_tobe_available(wid)
+                if (self.getWorkloadStatus(wid) == "available"):
+                    reporting.add_test_step(
+                        "Create workload using cli", tvaultconf.PASS)
+                else:
+                    reporting.add_test_step(
+                        "Create workload using cli", tvaultconf.FAIL)
+                    raise Exception("Create workload using cli failed.")
+            else:
+                reporting.add_test_step(
+                    "Create workload using cli", tvaultconf.FAIL)
+                raise Exception("Create workload using cli failed.")
+
+            # take full snapshot
+            snapshot_id = self.workload_snapshot(wid, True)
+
+            self.wait_for_workload_tobe_available(wid)
+            snapshot_status = self.getSnapshotStatus(wid, snapshot_id)
+
+            if (snapshot_status == "available"):
+                LOG.debug("Full snapshot created.")
+                reporting.add_test_step("Create full snapshot", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Create full snapshot", tvaultconf.FAIL)
+                raise Exception("Full snapshot creation failed.")
+
+            # take incremental snapshot
+            incr_snapshot_id = self.create_snapshot(wid, is_full=False)
```

Comment:
What is the difference between full and incremental snapshot here? There is no data update done on the volume side.

Reply:
It was there earlier but then I removed it because it was not serving any purpose as it is (incr_snapshot_id) only used in One-Click restore and there also we are not calculating any checksum to validate. 
We are just checking mount command status and for that purpose we don't need any additional data on the volume. 

Please correct me if I am wrong.  If you think we should still add more data before incremental snapshot I will add those code block again.